pkg/controller/scyllacluster/resource_test.go

```diff
@@ -925,3 +927,227 @@ func TestStatefulSetForRack(t *testing.T) {
+			services:          map[string]*corev1.Service{},
+			expectedIngresses: nil,
+		},
+		{
+			name: "ingress objects are generated for every domain",
+			services: map[string]*corev1.Service{
+				"any":    newIdentityService("any"),
+				"node-1": newMemberService("node-1", "host-id-1"),
+				"node-2": newMemberService("node-2", "host-id-2"),
+				"node-3": newMemberService("node-3", "host-id-3"),
+			},
+			cluster: func() *scyllav1.ScyllaCluster {
+				cluster := basicScyllaCluster.DeepCopy()
+				cluster.Spec.DNSDomains = []string{"public.scylladb.com", "private.scylladb.com"}
+				cluster.Spec.ExposeOptions = &scyllav1.ExposeOptions{
+					CQL: &scyllav1.CQLExposeOptions{
+						Ingress: &scyllav1.IngressOptions{
+							IngressClassName: "cql-ingress-class",
+							Annotations: map[string]string{
+								"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+							},
+						},
+					},
+				}
+
+				return cluster
+			}(),
+			expectedIngresses: []*networkingv1.Ingress{
+				newIngress("any-cql",
+					"AnyNode",
+					"cql-ingress-class",
+					map[string]string{
+						"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+					},
+					newIngressRule("any.cql.public.scylladb.com", "any", "cql-ssl", "/", networkingv1.PathTypePrefix),
+					newIngressRule("any.cql.private.scylladb.com", "any", "cql-ssl", "/", networkingv1.PathTypePrefix),
+				),
+				newIngress(
+					"node-1-cql",
+					"Node",
+					"cql-ingress-class",
+					map[string]string{
+						"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+					},
+					newIngressRule("host-id-1.cql.public.scylladb.com", "node-1", "cql-ssl", "/", networkingv1.PathTypePrefix),
+					newIngressRule("host-id-1.cql.private.scylladb.com", "node-1", "cql-ssl", "/", networkingv1.PathTypePrefix),
+				),
+				newIngress(
+					"node-2-cql",
+					"Node",
+					"cql-ingress-class",
+					map[string]string{
+						"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+					},
+					newIngressRule("host-id-2.cql.public.scylladb.com", "node-2", "cql-ssl", "/", networkingv1.PathTypePrefix),
+					newIngressRule("host-id-2.cql.private.scylladb.com", "node-2", "cql-ssl", "/", networkingv1.PathTypePrefix),
+				),
+				newIngress(
+					"node-3-cql",
+					"Node",
+					"cql-ingress-class",
+					map[string]string{
+						"my-cql-custom-annotation": "my-cql-custom-annotation-value",
+					},
+					newIngressRule("host-id-3.cql.public.scylladb.com", "node-3", "cql-ssl", "/", networkingv1.PathTypePrefix),
+					newIngressRule("host-id-3.cql.private.scylladb.com", "node-3", "cql-ssl", "/", networkingv1.PathTypePrefix),
+				),
+			},
+		},
+	}
+
+	for i := range tcs {
+		tc := tcs[i]
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			got := MakeIngresses(tc.cluster, tc.services)
+
+			less := func(v []*networkingv1.Ingress) func(i, j int) bool {
+				return func(i, j int) bool {
+					return v[i].Name < v[j].Name
+				}
+			}
+			sort.Slice(got, less(got))
```

Comment:
the function must have a consistent output, if not we should fix it there

Reply:
fixed